### PR TITLE
feat: post init hooks and project cache

### PR DESCRIPTION
**Pull request by doosuu**
Implements 
* post init hooks for components
* project based caching

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2023 Robert Bosch GmbH and Microsoft Corporation
```
Comment (dennismeister93) [inline .github/workflows/ci.yml:1]: we could merge the tests actions into the ci workflow file

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
```
Comment (dennismeister93) [inline .pre-commit-config.yaml:5]: in the other repos we are already using
rev: v4.4.0

**Review by dennismeister93 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (doosuu) [reply .github/workflows/ci.yml:1]: Good idea!

File: .pre-commit-config.yaml (hunk 2 above)
Comment (doosuu) [reply .pre-commit-config.yaml]: Done!

**Review by dennismeister93 — APPROVED** [review]
LGTM, as discussed, packages will be updated according to ENV name changes after cli update in base images

**Pull request merged**